src/renderers/WebGLRenderer.js

```diff
@@ -1447,6 +1445,11 @@ class WebGLRenderer {
 			const currentToneMapping = _this.toneMapping;
 			_this.toneMapping = NoToneMapping;
 
+			// Remove viewport from camera to avoid nested render calls resetting viewport to it (e.g Reflector).
+			// Transmission render pass requires viewport to match the transmissionRenderTarget.
+			const currentCameraViewport = camera.viewport;
```

Comment:
It's good that you added a comment that explains this bit. Maybe we can find a different XR viewport handling in the future that avoids the viewport reset in `Reflector` in the first place.

Reply:
@mrxz I understand it makes sense to define the viewport to define the size of the render target. However, is the temporary removal really necessary?

What can be irritating is that the viewport can be defined via render targets and `camera.viewport`. When a reflector is used in a XR scene _without_ transmissive objects, the viewport isn't correct after restoring the XR render target. `camera.viewport` holds the correct values. So after `renderer.setRenderTarget( currentRenderTarget );` at the end of `onBeforeRender()`, the reflector restores the viewport of the sub camera.

In context of transmission, the transmission render target should now have the same size as `camera.viewport`. So is the temporary removal of `camera.viewport` really necessary? When debugging the latest changes today, it seems to me these lines are redundant. If not, can you please explain what bit I am missing?

Reply:
Thanks! That helped quite a lot!